addons/stock/models/stock_orderpoint.py

```diff
@@ -1,21 +1,29 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import logging
+from collections import defaultdict
 from datetime import datetime, time
-from json import dumps
-
 from dateutil import relativedelta
+from itertools import groupby
+from json import dumps
+from psycopg2 import OperationalError
 
-from odoo import _, api, fields, models
+from odoo import SUPERUSER_ID, _, api, fields, models, registry
+from odoo.addons.stock.models.stock_rule import ProcurementException
 from odoo.exceptions import UserError, ValidationError
 from odoo.osv import expression
+from odoo.tools import float_compare, frozendict, split_every
+
+_logger = logging.getLogger(__name__)
 
 
 class StockWarehouseOrderpoint(models.Model):
     """ Defines Minimum stock rules. """
     _name = "stock.warehouse.orderpoint"
     _description = "Minimum Inventory Rule"
     _check_company_auto = True
+    _order = "location_id,company_id,id"
```

Comment:
indexing location_id would seems a good idea
maybe a composite with product_id to help the grouping ?
we'd need to test on a proper database